File: applicatioan-1/ogrencibilgisistemi-c#/OgrenciBilgiSistemi/BilgiPaketi.cs
using System;
using System.Collections.Generic;

public class BilgiPaketi
{
    private List<Ders> tumDersler;

    public BilgiPaketi()
    {
        tumDersler = new List<Ders>();
    }

    public void DersEkle(Ders ders)
    {
        tumDersler.Add(ders);
    }

    public void TranskriptSorgula(Ogrenci ogrenci)
    {
        Console.WriteLine(ogrenci.GetAdSoyad() + " Ders Listesi:");
        foreach (Ders ders in tumDersler)
        {
            if (ders.OgrenciKayitliMi(ogrenci.Numara))
            {
                if (ders is TemelDers)
                {
                    Console.WriteLine($"{ders.DersAdi} dersi, Kredisi: {ders.Kredi}, Harf Notu: {ders.NotHesapla(80, 70, 90)}");
                }
                else if (ders is UygulamaliDers)
                {
                    Console.WriteLine($"{ders.DersAdi} dersi, Kredisi: {ders.Kredi}, Harf Notu: {ders.NotHesapla(85, 75)}");
                }
                else if (ders is MeslekiEgitimDersi)
                {
                    Console.WriteLine($"{ders.DersAdi} dersi, Kredisi: {ders.Kredi}, Harf Notu: {ders.NotHesapla(95)}");
                }
            }
        }
    }
}

File: applicatioan-1/ogrencibilgisistemi-python/bilgi_paketi.py
class BilgiPaketi:
    def __init__(self):
        self.tum_dersler = []

    def ders_ekle(self, ders):
        self.tum_dersler.append(ders)

    def transkript_sorgula(self, ogrenci):
        print(f"{ogrenci.get_ad_soyad()} Ders Listesi:")
        for ders in self.tum_dersler:
            if ders.ogrenci_kayitli_mi(ogrenci.numara):
                try:
                    harf_notu = ders.not_hesapla(ogrenci)
                    print(f"{ders.ders_adi} dersi, Kredisi: {ders.kredi}, Harf Notu: {harf_notu}")
                except ValueError as e:
                    print(f"{ders.ders_adi} dersi için not girilmemiş.")
